### PR TITLE
Fix deprecations and build warnings

### LINKED ISSUE #9: Fix deprecations
g_type_class_add_private and friends + some others throw warnings

### PR DESCRIPTION
**Pull request by tari01**
fixes #9

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -3,14 +3,14 @@ cmake_minimum_required(VERSION 2.8.9)
 
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
 
-set(PROJECT_VERSION "2.1.0")
+set(PROJECT_VERSION "2.0.95")
```
Comment (sunweaver) [inline CMakeLists.txt:6]: URGH, why lower the version numer?

File: CMakeLists.txt [hunk 2]
```diff
@@ -69,7 +69,7 @@ if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
   set(C_WARNING_ARGS "${C_WARNING_ARGS} -Wno-c++98-compat -Wno-padded") # these are annoying
   set(C_WARNING_ARGS "${C_WARNING_ARGS} -Wno-documentation") # gtk-doc != doxygen
 else()
-  set(C_WARNING_ARGS "${C_WARNING_ARGS} -Wall -Wextra -Wpedantic -Wformat=2")
+  set(C_WARNING_ARGS "${C_WARNING_ARGS} -Wall -Wformat=2") # Use -Wextra and -Wpedantic explicitly if needed
```
Comment (sunweaver) [inline CMakeLists.txt:72]: Why did you remove these two. I find it actually quite useful seeing as many warnings as possible during nightly builds.

**Review by sunweaver — COMMENTED** [review]
The version number needs to be fixed. Please provide feedback about the compiler flags -Wextra and -Wpedantic and why you removed those. The rest looks ok.

File: CMakeLists.txt (hunk 2 above)
Comment (tari01) [reply CMakeLists.txt:72]: They spit out a whole lot of warnings that cannot and need not be really fixed (for properly done type casting, etc). I think we should bear in mind that our primary build audience are users of all sorts of distros needing to build these components for themselves. It makes sense to provide them with a nice clean build experience, and explicitly enable overkill warning mode for development builds.

File: CMakeLists.txt (hunk 1 above)
Comment (tari01) [reply CMakeLists.txt:6]: ANTI-URGH: I saw that, but it wasn't me! I always fetch+rebase upstream before making changes. It either came from master or my cmake did it...

Comment (sunweaver) [reply CMakeLists.txt:6]: @tari01: Hmmm... it must not be in this PR, then. I will amend the commit before merging.
